### ISSUE TITLE
Don't ignore :<>= even when they are invalid comparators

### ISSUE THREAD

**Issue by NilsEnevoldsen** [scryfall/feedback#697]
1. A query for `m*@&#$u` is equivalent to a query for `mu`.
2. A query for `m*@&:#$u` is equivalent to a query for `mu`.
3. A query for `m*@&<#$u` is equivalent to a query for `mu`.
4. A query for `m*@&>#$u` is equivalent to a query for `mu`.
5. A query for `m*@&=#$u` is equivalent to a query for `mu`.

While 1 is clearly a deliberate design decision, 2-5 can lead users into traps; consider searches for `m<u`, `m>u`, and `m=u`. (I'm not actually sure if `:` could ever mislead a user, but on principle it falls into the same bucket.)

I suggest invalidating _all_ expressions that include `:<>=` (as is done for, say, `w:u`) instead of stripping them out and doing a name search.

**Comment by spacemonaut** [comment]
There are cards with `:` in their name though: https://scryfall.com/search?q=name%3A%2F%3A%2F

The scenarios you've described don't seem very realistic, what scenarios have you actually bumped into that would be improved by failing fast? Developers in projects like these can usually do best — and do their best user experience design — when they're given real scenarios that happened and which they can understand how to resolve, rather than given a specific solution they should implement to solve it.

**Comment by NilsEnevoldsen** [comment]
Yes, that's all true. Thanks for the feedback @spacemonaut!

I filed this after searching for something like `m<wub`. Something similar has happened with another keyword (maybe regex related?), although I don't remember what it was.

I retract my suggestion, and I'll just replace it with the narrower issue that the `mana:` keyword doesn't work as expected with `<>=`.

**Comment by Sniffnoy** [comment]
This of course brings up the question: What *is* the expected the expected behavior here?  I can think of two possibilities.  (This question has come up here before and... well, I cite earlier stuff below.)

One is that it works like magiccards.info, where A<=B means, essentially, "A is contained in B".  You have most of this already!  The only real problem is that right now you directly check whether A is contained in B without accounting for the fact that that doesn't make sense for generic mana symbols -- one should have {1}{G}<={2}{G}, rather than saying, ah, well, {1} and {2} are different symbols, so these are incomparable.  The symbol {N} should be treated as shorthand for N {1}s.

(Well, there's that problem and there's fact that right now you can effectively only do mana>=A searches, and not <= or = searches, but that's another matter.)

The other possibility is the crazier one I floated in #481, so that e.g. one would have {B/G}<={G}.

(IMO both would be useful, but coming up with a syntax that allows either sounds a little ridiculous...)

**Comment by csuhta** [comment]
We have so far avoided implementing mana cost range comparisons because there is no strict definition in the game rules when comparing mana costs, the game only compares *converted* mana cost. There are a lot of edge cases:

- It's fairly clear that `{4}{G}` > `{3}{G}` 
- But `{G}` ?= `{1}` 
- What about `{1}{G}` ?= `{2}{G}{G}`
- And also `{G}` ?= `{G/P}` 
- Hrmmm `{2/G}` ?= `{2}` 
- Hrmmm `{C}` ?= `{1}` 
- Uhhh `{G}{G}` ?= `{R}{G}`
- Uhhh `{G}{G}{G}{G}{G}` ?= `{2}{G}`
- Uhhhh `{2}{W/U}` ?= `{1}{W}{U}`
- Uhhhhhh `{0}` ?= (no mana cost)

We also did some extensive testing with MCI and there are a lot of situations that don't appear to be working correctly, so we don't have a reference implementation:

- https://magiccards.info/query?q=mana%3C3G&v=card&s=cname&s=scan (Cards that can't be cast appear)
- https://magiccards.info/query?q=mana%3C%3DGGGGGG+cmc%3E1&s=cname&v=scan&p=1 (excludes cards with colorless parts or any other color for some reason)

Even if we made a call one way or another on each of these questions, 50% of the people submitting feedback will not agree with what we've decided

**Comment by Sniffnoy** [comment]
Seems to me that 50% of the people submitting feedback disagreeing would be an improvement over the current state, where the handling of generic costs means that nobody agrees!  I mean you may as well just remove mana cost searching entirely if you want to avoid complaints, right?  Because right now even the easy cases are wrong and the result is just seriously confusing.

I mean all the examples that you posted are pretty easy to resolve if you pick one of the two partial orders and document it... I was going to include a table to demonstrate except I guess you already know what it would contain.  The one exception is no mana cost, something I'll admit I'd failed to consider.  Under the "easier to cast" partial order I guess that should be the maximum possible cost, except that that would be confusing in searches and so for practical purposes it probably makes more sense to consider it as incomparable to everything else, which is probably also the practical choice under the containment partial order.

Anyway yeah point is current method is worse than either (and already close to one of them) so changing it doesn't seem like it could make matters *worse*?

**Comment by NilsEnevoldsen** [comment]
I have to agree with Sniffnoy. The current logic is bizarre, as far as in-game significance goes.

> all the examples that you posted are pretty easy to resolve if you pick one of the two partial orders and document it

The system described in #481 seems clearly preferable, although I can't speak to the computational tractability.
